pkg/apis/authorization/types.go

```diff
@@ -144,3 +144,53 @@ type SubjectAccessReviewStatus struct {
 	// For instance, RBAC can be missing a role, but enough roles are still present and bound to reason about the request.
 	EvaluationError string
 }
+
+// +genclient=true
+// +noMethods=true
+
+// SelfSubjectRulesReview enumerates the set of actions the current user can perform within a namespace.
+// The returned list of actions may be incomplete depending on the server's authorization mode,
+// and any errors experienced during the evaluation.
+type SelfSubjectRulesReview struct {
```

Comment:
One thing we've learned from SARs is that users may have information about themselves that we do not.  In particular, they may have information about the `user.Extra` bits.  I'd like to see this API take that into account inside of a `Spec` stanza that allows clearing of `user.Extra`, doing nothing, or setting a specific `user.Extra` (no merging).